pkg/handler/common/provider/kubevirt.go

```diff
@@ -224,3 +232,87 @@ func KubeVirtStorageClassesWithClusterCredentialsEndpoint(ctx context.Context, u
 
 	return KubeVirtStorageClasses(ctx, kvKubeconfig)
 }
+
+func filterVMIPresets(vmiPresets apiv2.VirtualMachineInstancePresetList, quota kubermaticv1.MachineDeploymentVMResourceQuota) apiv2.VirtualMachineInstancePresetList {
+	filteredVMIPresets := apiv2.VirtualMachineInstancePresetList{}
+
+	// Range over the records and apply all the filters to each record.
+	// If the record passes all the filters, add it to the final slice.
+	for _, vmiPreset := range vmiPresets {
+		presetSpec := kubevirtv1.VirtualMachineInstancePresetSpec{}
+		if err := json.Unmarshal([]byte(vmiPreset.Spec), &presetSpec); err != nil {
+			log.Logger.Errorf("skipping VMIPreset:%s, parsing preset.Spec failed:%v", vmiPreset.Name, err)
+			continue
+		}
+
+		cpu, memory, err := GetKubeVirtPresetResourceDetails(presetSpec)
+		if err != nil {
+			log.Logger.Errorf("skipping VMIPreset:%s, fetching presetResourceDetails failed:%v", vmiPreset.Name, err)
+			continue
+		}
+
+		if handlercommon.FilterCPU(int(cpu.AsApproximateFloat64()), quota.MinCPU, quota.MaxCPU) && handlercommon.FilterMemory(int(memory.Value()/(1<<30)), quota.MinRAM, quota.MaxRAM) {
+			filteredVMIPresets = append(filteredVMIPresets, vmiPreset)
+		}
+	}
+	return filteredVMIPresets
+}
+
+// GetKubeVirtPresetResourceDetails extracts cpu and mem resource requests from the kubevirt preset
+// for CPU, take the value by priority:
+// - check if spec.cpu is set, if socket and threads are set then do the calculation, use that
+// - if resource request is set, use that
+// - if resource limit is set, use that
+// for memory, take the value by priority:
+// - if resource request is set, use that
+// - if resource limit is set, use that.
+func GetKubeVirtPresetResourceDetails(presetSpec kubevirtv1.VirtualMachineInstancePresetSpec) (resource.Quantity, resource.Quantity, error) {
```

Comment:
what about the default `100mi` CPU and `1Gi` memory? Why dont we have this?

Reply:
After all we decided to skip presets that they don't have it. We require from presets that those are set.

Reply:
Somehow my eyes just went over the check which throws the error when nothing is set. Looks good